### DISCUSSION TITLE
Hook FastLogin into my plugin

### DISCUSSION THREAD

**Discussion by iKstruuh** [games647/FastLogin#706] [Q&A] [answered]
Hi, to summarize, is there a way to hook FastLogin into my plugin? For example, check if a player is using /premium to make other stuff with my plugin?

**Comment by TuxCoding** [comment]
There is https://github.com/games647/FastLogin/blob/6f16700cdf055ee67a5e2318379ae86641f9957d/bukkit/src/main/java/com/github/games647/fastlogin/bukkit/FastLoginBukkit.java#L251

to query the current premium status if that is what you want.

**Reply by iKstruuh** [reply]
Isn't there a method that returns true or false?
To check if the player is registered as a premium player or not in FastLogin?

And how can I implement FastLogin into my plugin?
Which class have I to copy?

**Comment by TuxCoding** [comment]
It's an enum representing three states instead of two. You don't need to copy anything. Access the plugin instance using `JavaPlugin.getPlugin` and then you can call that method.

**Reply by iKstruuh** [reply]
`PREMIUM` means that the player is premium, but what do `CRACKED` and `UNKNOWN` mean?

**Comment by TuxCoding** [comment, marked as answer]
It represents how the player logged in. If it was premium connection or not.

**Reply by iKstruuh** [reply]
Thank you!

**Reply by iKstruuh** [reply]
I think that my issue is now solved!
